modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/MappedFragment.java

```diff
@@ -66,10 +71,30 @@ public class MappedFragment {
         this.target = target;
     }
 
+    /** Constructor. */
+    private MappedFragment(
+            Fragment fragment,
+            List<ColocationGroup> groups, List<String> nodes,
+            Long2ObjectMap<ColocationGroup> groupsBySourceId,
+            @Nullable Long2ObjectMap<List<String>> sourcesByExchangeId,
+            @Nullable ColocationGroup target
+    ) {
+        this.fragment = fragment;
+        this.nodes = List.copyOf(nodes);
+        this.groupsBySourceId = groupsBySourceId;
+        this.sourcesByExchangeId = sourcesByExchangeId;
```

Comment:
I think we should find a better name for `sourcesByExchangeId ` fields. Because it is not obvious that it stores node name (yes, it stores string which sorta-kinda looks like node names) but still.